Tasks/Maven/maventask.ts

```diff
@@ -258,6 +265,35 @@ mvnv.exec()
     console.error("SonarQube analysis failed");
     sqRunFailed = true;
 })
+.then(function (code) { // Pick up code analysis file(s)
+    var analysisResults: ar.AnalysisResult[] = [];
+
+    // PMD
+    if (tl.getInput('pmdAnalysisEnabled', true) == 'true') {
+        var pmdResults:ar.AnalysisResult = pmd.processPmdOutput(tl.getVariable('build.sourcesDirectory'));
+        if (pmdResults) {
+            analysisResults.push(pmdResults);
+        }
+    }
+
+    // Process analysis results - upload files and generate build summary lines
+    var buildSummary:string = '';
+    analysisResults.forEach(analysisResult => {
+        buildSummary += buildSummaryLineFromResult(analysisResult) + "  \r\n";
+    });
+
+    // Save and upload build summary
+    var buildSummaryFilePath:string = path.join(tl.getVariable('build.sourcesDirectory'), "target", "buildsummary.md");
```

Comment:
maybe rename that file to CodeAnalysisSummary.md, in the (hightly unlikely) event of a name clash.